### ISSUE TITLE
pillow 4.2.1 regression with font.getsize on empty strings

### ISSUE THREAD

**Issue by akrherz** [conda-forge/pillow-feedstock#34]
My apologies if this is an inappropriate bug to report here, but the release of 4.2.1 Pillow onto conda-forge has a regression that hit my code hard.  The regression is logged here: python-pillow/Pillow#2614

1. Is it appropriate to ask for the conda-forge version to be rolled back or
2. make a PR with that patch included that bumps the conda-forge version or
3. these types of conda-forge / bleeding-edge breakages are expected and just need to wait for next upstream release?

thank you for your time and making this available for conda.

**Comment by ocefpaf** [comment]
The best options are:

> Is it appropriate to ask for the conda-forge version to be rolled back or

Not really b/c this is ultimately an upstream issue but you can always pin your workflow to the previous working version. This is usually the fastest b/c it depends only on things you can control (most of the times :smile: ).

> make a PR with that patch included that bumps the conda-forge version or

That is a more involved solution but probably better. If you have the time please do so!

> these types of conda-forge / bleeding-edge breakages are expected and just need to wait for next upstream release?

`4.2.1` is an official release so I would not call "bleeding-edge breakages." Unfortunately even "stable" releases causes breakages though.

**Comment by akrherz** [comment]
@ocefpaf Thank you very much for the detailed response.  I don't have enough skill to properly generate and test a PR for this.  If you want to close this or just leave it open until upstream releases `4.2.2`, that would be fine by me.

**Comment by ocefpaf** [comment]
Do you have a small image/script that reproduces the problem?

**Comment by akrherz** [comment]
@ocefpaf not really, the issue is already fixed upstream.  The reproducer is simply to call `font.getsize` on an empty string.

**Comment by ocefpaf** [comment]
Got it. See #35

**Comment by akrherz** [comment]
oh wow, providing patches is a lot more automated than I thought.  Thank you kindly for doing this.

**Comment by ocefpaf** [comment]
There is more there than only the patch. It is not a "clean" PR.

The "correct" way to do this would be to submit the patch, bump the build number, and add a test.
I did not do the latter b/c it is tested upstream. And the PR contains a change in our pinning policies and a re-render.

TL;DR just the patch would be much easier :smile:

**Comment by akrherz** [comment]
Just to note that the newly released 4.3.0 Pillow still has an issue closely related to this one. see python-pillow/Pillow#2783

**Comment by ocefpaf** [comment]
Thanks for the heads up!

**Comment by hugovk** [comment]
The two issues:

* `ImageFont.font.getsize("")` has been fixed in this week's Pillow 4.3.0 release (https://github.com/python-pillow/Pillow/issues/2614), 
* `ImageDraw.textsize("")` has just been fixed in latest master (https://github.com/python-pillow/Pillow/issues/2783).
